r/src/compute-exec.cpp

```diff
@@ -141,7 +152,8 @@ class ExecPlanReader : public arrow::RecordBatchReader {
   std::shared_ptr<arrow::Schema> schema_;
   std::shared_ptr<arrow::compute::ExecPlan> plan_;
   arrow::AsyncGenerator<std::optional<compute::ExecBatch>> sink_gen_;
-  int status_;
+  enum ExecPlanReaderStatus status_;
```

Comment:
In C++ you probably don't need to specify `enum`.
```suggestion
  ExecPlanReaderStatus status_;
```

Reply:
By the way, for readability I would call it something else than `status_`, which is usually a `arrow::Status` instance...